### ISSUE TITLE
Unable to select a kernel when kernel has exact same name

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode-jupyter#361]
Assume you have two kernels with the same name, its impossible to select the second kernel.
This is possible with Conda environments. 
**With Conda, the kernels get registered under each environment.**
We know this today, however we have made the assumption that kernel names & display names are unique everywhere.

This is not true.
With a conda environment we can can a Python3 kernel in the conda environment as well.
So we have Python3 kernel in root as well as Conda environments & display names are the same as well.

**Here's how I came across this**
* Install beakerx (name & display name = `java`)
* Install IJava (name & display name = `java`)
* Now its impossible to select the ijava kernel as we always use `beakerx`.

**Same issues with Python3 kernels under conda**

**Temporary work around is to update the display names (which requires code changes)**
* Manually edit kernel.json files & ensure display names are unique (PR submitted to get this working)
* This still didn't work as we have a weird appraoch of starting kernels.
* Basically even though i have picked the kernel spec we end up searching for kernel specs all over again in lower layers.
	* That too didn't work, as we were searching everywhere only based on kernel name
	* Work around PR submitted

The root cause of the bug is the lower layers of the code doing stuff that's unnecessary (we have all of the info we need).
**Once we have a kernel spec, we should just use that as it is**
